### PR TITLE
Incorrect processing JSON RPC responses

### PR DESCRIPTION
**Pull request by dkulieshov**
Signed-off-by: Dmitry Kuleshov <dkuleshov@codenvy.com>

### What does this PR do?
Fixes situations where we do not define actions for responses when sending requests over Json RPC

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/5135

#### Changelog
Fixed response handling if no response action is set

#### Release Notes
Fixed response handling if no response action is set

### CODE REVIEW

File: core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/ResponseDispatcher.java [hunk 1]
```diff
@@ -108,9 +116,17 @@ public void dispatch(String endpointId, JsonRpcResponse response) {
     private void processError(String endpointId, JsonRpcResponse response, String key) {
         LOGGER.debug("Response has error. Proceeding...");
 
+        if (!promises.containsKey(key)) {
+            LOGGER.debug("No action is associated for the key: " + key + ", skipping.");
```
Comment (benoitf) [inline core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/ResponseDispatcher.java]: AFAIK you should not use string concatenation in debug traces.

**Review by RomanNikitenko — APPROVED** [review]

File: core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/ResponseDispatcher.java [hunk 2]
```diff
@@ -119,14 +135,31 @@ private void processError(String endpointId, JsonRpcResponse response, String ke
...
+
+        if (!rClasses.containsKey(key)) {
+            LOGGER.debug("No result class is associated for the key: {}, skipping.", key);
+            return;
+        }
+
         JsonRpcResult result = response.getResult();
+
+        Class<?> rClass = rClasses.remove(key);
+        LOGGER.debug("Fetching result class: {}",rClass);
```
Comment (benoitf) [inline core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/ResponseDispatcher.java]: there is a missing space after the comma so maybe formatter was not applied ?

File: core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/ResponseDispatcher.java [hunk 3]
```diff
@@ -108,9 +116,17 @@ public void dispatch(String endpointId, JsonRpcResponse response) {
...
+            LOGGER.debug("No action is associated for the key: {}, skipping.", key);
+            return;
+        }
+
         JsonRpcError error = response.getError();
-        JsonRpcPromise<JsonRpcError> jsonRpcPromise = cast(promises.remove(key));
-        BiConsumer<String, JsonRpcError> consumer = jsonRpcPromise.getFailureConsumer();
+
+        JsonRpcPromise<?> promise = promises.remove(key);
+        LOGGER.debug("Fetching promise:" + promise);
```
Comment (benoitf) [inline core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/ResponseDispatcher.java]: I still see a concatenation there

**Review by vparfonov — APPROVED** [review]

**Review by svor — APPROVED** [review]

**Comment by benoitf** [comment]
@dkuleshov this PR needs to be rebased in order to be merged.
BTW what was blocking the merge of this PR as it was accepted since a long time ?

**Review by vparfonov — APPROVED** [review]

**Comment by dkulieshov** [comment]
This is already in master under another PR https://github.com/eclipse/che/pull/5341

**Pull request closed without merging**